Content.Server/_DV/Abilities/Chitinid/ChitinidSystem.cs

```diff
@@ -0,0 +1,123 @@
+using Content.Server.Abilities.Felinid;
+using Content.Server.Chat.Systems;
+using Content.Server.Nutrition.Components;
+using Content.Server.Popups;
+using Content.Shared.Actions;
+using Content.Shared.Actions.Events;
+using Content.Shared.Audio;
+using Content.Shared.Damage;
+using Content.Shared.Damage.Prototypes;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Inventory;
+using Content.Shared.Mobs;
+using Content.Shared.Mobs.Systems;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Abilities.Chitinid;
+
+public sealed partial class ChitinidSystem : EntitySystem
+{
+    [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+    [Dependency] private readonly InventorySystem _inventorySystem = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly DamageableSystem _damageable = default!;
+    [Dependency] private readonly MobStateSystem _mobState = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<ChitinidComponent, ComponentInit>(OnInit);
+        SubscribeLocalEvent<ChitinidComponent,ChitziteActionEvent>(OnChitzite);
+        SubscribeLocalEvent<ChitinidComponent,MapInitEvent>(OnMapInit);
+    }
+
+    private Queue<EntityUid> RemQueue = new();
+
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+        foreach (var bug in RemQueue)
+        {
+            RemComp<CoughingUpChitziteComponent>(bug);
+        }
+        RemQueue.Clear();
+
+        var query = EntityQueryEnumerator<ChitinidComponent, DamageableComponent>();
+        while (query.MoveNext(out var uid, out var chitinid, out var damageable))
+        {
+            if (_gameTiming.CurTime < chitinid.NextUpdate)
+                continue;
+
+            chitinid.NextUpdate += chitinid.UpdateInterval;
+
+            if (chitinid.AmountAbsorbed >= chitinid.MaximumAbsorbed && !_mobState.IsDead(uid))
```

Comment:
replace `&& !` with `||`, this currently makes dead chitinid regen radiation forever